### ISSUE TITLE
questions about vgg-ssd in uniform-tf

### ISSUE THREAD

**Issue by wangxianrui** [Tencent/PocketFlow#206]
**Describe the bug**
in the eval function of uniform-tf, there is an error said that `global_step is uninitialled`, and i try to find this is because we create global_step in `calc_loss_fn` of vgg_at_pascalvoc.py,[link](https://github.com/Tencent/PocketFlow/blob/master/nets/vgg_at_pascalvoc.py#L405) and don't use the pretrained value to recover it.  
and we can solve it by add global_step variable to the eval_saver in uniform_quantization_tf/learner.py [link](https://github.com/Tencent/PocketFlow/blob/master/learners/uniform_quantization_tf/learner.py#L305)

and another question, why use `loss_w_cls` as a coefficient about `ce_loss` in `calc_loss_fn`,[link](https://github.com/Tencent/PocketFlow/blob/master/nets/vgg_at_pascalvoc.py#L406) is there theory evidence to prove it useful? or some documents about it? i have not saw it before.

**Comment by wangxianrui** [comment]
and then, the variable added to collections in [here](https://github.com/Tencent/PocketFlow/blob/master/learners/uniform_quantization_tf/learner.py#L315) should be 'logits['predictions']' which we want to get in ssd eval, but it is also a dict, so we should do some preparation before add it to collections.

**Comment by wangxianrui** [comment]
@jiaxiang-wu

**Comment by jiaxiang-wu** [comment]
1. Yes, this is a bug. We need to use the pre-trained value to recover `global_step`.
2. This is a small trick that we discovered to be useful when using `MomentumOptimizer` to fine-tune the quantized network. However, later on, we switch to `AdamOptimizer` and whether this trick is still needed is not fully tested.
3. What do you mean by "we should do some preparation before add it to collections"?

**Comment by wangxianrui** [comment]
the outputs of forward_fn include 'cls_pred' 'loc_pred' and 'predictions', and 'cls_pred' 'loc_pred' are used to calculate loss in train,  'predictions' are bboxes and scores which we want in model's eval  stage. but 'predictions' are type of 'dict', we can't add it directly to collections in tensorflow.

**Comment by jiaxiang-wu** [comment]
The `add_to_collection` operation is used to mark the output tensor(s) for exporting *.pb & *.tflite models. What's your opinion on which tensor(s) should be the output tensor(s) in *.pb & *.tflite models, `cls_pred`, `loc_pred`, or items in `predictions`?

**Comment by wangxianrui** [comment]
I think it should be the `predictions`.
`cls_pred` and  'loc_pred' are used to calculate loss in train age [here](https://github.com/Tencent/PocketFlow/blob/master/nets/vgg_at_pascalvoc.py#L285) 
'predictions' are the final result used to dump to txt [here](https://github.com/Tencent/PocketFlow/blob/master/nets/vgg_at_pascalvoc.py#L553)

**Comment by jiaxiang-wu** [comment]
Okay, then we need to find out a way to pack all the values in `predictions`.
